src/lib/transaction_logic/parties_logic.ml

```diff
@@ -864,14 +864,22 @@ module Make (Inputs : Inputs_intf) = struct
        verify a snapp proof.
     *)
     Account.register_verification_key a ;
-    let predicate_satisfied : Bool.t =
+    let predicate_satisfied =
       h.perform (Check_predicate (is_start', party, a, global_state))
     in
-    let protocol_state_predicate_satisfied : Bool.t =
+    let local_state =
+      Local_state.add_check local_state Account_precondition_unsatisfied
+        predicate_satisfied
+    in
+    let protocol_state_predicate_satisfied =
       h.perform
         (Check_protocol_state_predicate
            (Party.protocol_state party, global_state))
     in
+    let local_state =
+      Local_state.add_check local_state Protocol_state_precondition_unsatisfied
```

Comment:
Can you remove the checks for these below (c. L1241)? No need to double-count them.